### ISSUE TITLE
Update existing Rubicon Scikit-learn pipelines to have multiple score metrics

### ISSUE THREAD

**Issue by shania-m** [capitalone/rubicon-ml#176]
**Is your enhancement request related to a problem? Please describe**

Currently Rubicon only allows a pipeline to log a single `score` metric to an experiment once. If a user needs to score across two different data sets, the first `score` will be recorded and the second call to score will result in an error: `A metric with name 'score' already exists`. 

**Describe the solution you'd like**
Rubicon users should be able to call `score` multiple times, and have a metric logged for each `score` generated.
For example a user should be able to define a Scikit-learn pipeline, fit the pipeline, and then `score` the accuracy across different data sets like this:

````

from rubicon_ml import Rubicon
from rubicon_ml.sklearn import FilterEstimatorLogger, RubiconPipeline


root_dir = os.environ.get("RUBICON_ROOT", "rubicon-root")
root_path = f"{os.path.dirname(os.getcwd())}/{root_dir}"

rubicon = Rubicon(persistence="filesystem", root_dir=root_path)
project = rubicon.get_or_create_project("Permutation Importance vs Random Forest Feature Importance ")

rf = RubiconPipeline(
    project,
    [
        ("preprocess", preprocessing),
        ("classifier", RandomForestClassifier(random_state=42)),
    ]
)

rf.fit(X_train, y_train)

print("RF train accuracy: %0.3f" % rf.score(X_train, y_train))
print("RF train accuracy: %0.3f" % rf.score(X_test, y_test))

````
This should result in the both scores being logged and printed without error.  
Recommendations: 
To address this issue, [rubicon sk-learn pipelines ](https://github.com/capitalone/rubicon-ml/blob/main/rubicon_ml/sklearn/pipeline.py) could explore generating a new experiment for each `score()` called. However, if a user would like all the scores to be linked to an experiment, we could also provide parameter to the function definition like so [`def score(self, X, y=None, sample_weight=None, experiment=None):`](https://github.com/capitalone/rubicon-ml/blob/2d1eaaa3b7475c968a9263f6bb7453d5dfc8c79a/rubicon_ml/sklearn/pipeline.py#L102) . 

Furthermore, we should explore [score_samples](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html#sklearn.pipeline.Pipeline.score_samples) and see if Rubicon can leverage this for multiple scoring.

**Additional context**
This Scikit-learn pipeline MDI [example](https://scikit-learn.org/stable/auto_examples/inspection/plot_permutation_importance.html) demonstrates how a pipeline fitted on training data may be used to leverage a score on training data accuracy and test data accuracy sequentially. Storing these scores allows the developer to draw conclusions on the models perfomance in training vs test.
